//
// PROJECT:         Aspia
// FILE:            category/category_dmi_pointing_device.proto
// LICENSE:         GNU Lesser General Public License 2.1
// PROGRAMMERS:     Dmitry Chapyshev (dmitry@aspia.ru)
//

syntax = "proto3";

option optimize_for = LITE_RUNTIME;

package aspia.proto;

message DmiPointingDevice
{
    enum Type
    {
        TYPE_UNKNOWN        = 0;
        TYPE_OTHER          = 1;
        TYPE_MOUSE          = 2;
        TYPE_TRACK_BALL     = 3;
        TYPE_TRACK_POINT    = 4;
        TYPE_GLIDE_POINT    = 5;
        TYPE_TOUCH_PAD      = 6;
        TYPE_TOUCH_SCREEN   = 7;
        TYPE_OPTICAL_SENSOR = 8;
    }

    enum Interface
    {
        INTERFACE_UNKNOWN             = 0;
        INTERFACE_OTHER               = 1;
        INTERFACE_SERIAL              = 2;
        INTERFACE_PS_2                = 3;
        INTERFACE_INFRARED            = 4;
        INTERFACE_HP_HIL              = 5;
        INTERFACE_BUS_MOUSE           = 6;
        INTERFACE_ADB                 = 7;
        INTERFACE_BUS_MOUSE_DB_9      = 8;
        INTERFACE_BUS_MOUSE_MICRO_DIN = 9;
        INTERFACE_USB                 = 10;
    }

    message Item
    {
        Type device_type           = 1;
        Interface device_interface = 2;
        int32 button_count         = 3;
    }

    repeated Item item = 1;
}
